src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http3RequestStream.cs

```diff
@@ -172,15 +172,10 @@ public async Task<HttpResponseMessage> SendAsync(CancellationToken cancellationT
                     await FlushSendBufferAsync(endStream: _request.Content == null, _requestBodyCancellationSource.Token).ConfigureAwait(false);
                 }
 
-                Task sendRequestTask;
-                if (_request.Content != null)
-                {
-                    sendRequestTask = SendContentAsync(_request.Content!, _requestBodyCancellationSource.Token);
-                }
-                else
-                {
-                    sendRequestTask = Task.CompletedTask;
-                }
+                // The return value indicates whether we've sent any content / whether we should wait for WritesClosed.
```

Comment:
We should wait for `WritesClosed` even without content (headers only request like GET).